test/lightning/maintenance/partition_table_service_test.exs

```diff
@@ -0,0 +1,495 @@
+        ],
+        [
+          "#{parent}_2023_03",
+          "FOR VALUES FROM ('2023-03-01 00:00:00') TO ('2023-03-31 00:00:00')"
+        ],
+        [
+          "#{parent}_2023_04",
+          "FOR VALUES FROM ('2023-04-01 00:00:00') TO ('2023-04-30 00:00:00')"
+        ],
+        [
+          "#{parent}_2023_05",
+          "FOR VALUES FROM ('2023-05-01 00:00:00') TO ('2023-05-31 00:00:00')"
+        ]
+      ]
+    end
+  end
+
+  describe "drop_empty_partition" do
+    test "drops the named partition" do
+      parent = "work_orders"
+
+      drop_range_partitions(parent)
+
+      add_partitions(parent)
+
+      expected =
+        [
+          "#{parent}",
+          "#{parent}_2023_01",
+          "#{parent}_2023_03",
+          "#{parent}_default",
+          "#{parent}_monolith"
+        ]
+        |> Enum.sort()
+
+      Service.drop_empty_partition(parent, "#{parent}_2023_02")
+
+      assert associated_relations(all_relations(), parent) == expected
+    end
+
+    test "does nothing if the table is not empty" do
+      parent = "work_orders"
+
+      drop_range_partitions(parent)
+
+      add_partitions(parent)
+
+      expected =
+        [
+          "#{parent}",
+          "#{parent}_2023_01",
+          "#{parent}_2023_02",
+          "#{parent}_2023_03",
+          "#{parent}_default",
+          "#{parent}_monolith"
+        ]
+        |> Enum.sort()
+
+      insert(:workorder, inserted_at: ~U[2023-02-15 10:00:00Z])
+
+      Service.drop_empty_partition(parent, "#{parent}_2023_02")
+
+      assert associated_relations(all_relations(), parent) == expected
+    end
+
+    test "errors out if the parent contains unexpected chars" do
+      parent = "work_orders"
+
+      drop_range_partitions(parent)
+
+      add_partitions(parent)
+
+      assert_raise(
+        ArgumentError,
+        fn ->
+          Service.drop_empty_partition("#{parent} --", "#{parent}_2023_02")
+        end
+      )
+
+      expected =
+        [
+          "#{parent}",
+          "#{parent}_2023_01",
+          "#{parent}_2023_02",
+          "#{parent}_2023_03",
+          "#{parent}_default",
+          "#{parent}_monolith"
+        ]
+        |> Enum.sort()
+
+      assert associated_relations(all_relations(), parent) == expected
+    end
+
+    test "errors out if the partition contains unexpected chars" do
+      parent = "work_orders"
+
+      drop_range_partitions(parent)
+
+      add_partitions(parent)
+
+      assert_raise(
+        ArgumentError,
+        fn ->
+          Service.drop_empty_partition(parent, "#{parent}_2023_2 --")
+        end
+      )
+
+      expected =
+        [
+          "#{parent}",
+          "#{parent}_2023_01",
+          "#{parent}_2023_02",
+          "#{parent}_2023_03",
+          "#{parent}_default",
+          "#{parent}_monolith"
+        ]
+        |> Enum.sort()
+
+      assert associated_relations(all_relations(), parent) == expected
+    end
+  end
+
+  defp all_relations() do
```

Comment:
I ended up with a fair number of functions to manage the fixtures - so of which have overlaps. If the overlaps are resolved I am hoping that I could reduce the number of fixture methods.

However, I will need to extend (at least) the scenarios that cover `:all` use case as I partition the other tables - as such I was hoping to delay the cleanup until those scenarios have been catered for.

Reply:
Thats cool, if the functions end up living in this test file thats all good imo.